### PR TITLE
bump windows build for vc 14.2 / vs 16 2019, add macos 11 build for py3.9

### LINKED ISSUE #1601: Update `windows-2016` to `windows-latest` and set up Visual Studio 2019 build
According to [https://github.com/actions/virtual-environments/issues/4312](https://github.com/actions/virtual-environments/issues/4312), the `windows-2016` environment will be deprecated on November 15, 2021 and removed on March 15, 2021. Our `perspective-python` build does not seem to be set up for Visual Studio 2019, as evidenced by build results from the [windows-latest](https://dev.azure.com/finosfoundation/perspective/_build/results?buildId=3496&view=logs&j=bfe8a88d-3928-550a-8c7c-c7cdeca2abfc&t=4da635ed-60f6-59ed-2d19-ec6821e7987b) branch.

There's some time to go before the image is removed, but it should be fixed before then.

### PR DESCRIPTION
**Pull request by timkpaine**
fixes #1601

### CODE REVIEW

**Comment by timkpaine** [comment]
can squash please

File: azure-pipelines.yml [hunk 1]
```diff
@@ -423,6 +435,24 @@ jobs:
       - ${{ parameters.pythonCoverageSteps }}
       - ${{ parameters.pythonMacWheelSteps }}
 
+  - job: "MacOS_BigSur"
```
Comment (texodus) [inline azure-pipelines.yml:438]: 🥇

File: python/perspective/setup.py [hunk 2]
```diff
@@ -148,10 +148,13 @@ def build_extension_cmake(self, ext):
         if platform.system() == "Windows":
             import distutils.msvccompiler as dm
 
+            # https://wiki.python.org/moin/WindowsCompilers#Microsoft_Visual_C.2B-.2B-_14.0_with_Visual_Studio_2015_.28x86.2C_x64.2C_ARM.29
```
Comment (texodus) [inline python/perspective/setup.py:151]: #tomatofarmer2022

**Review by texodus — APPROVED** [review]
Thanks for the PR!  Looks good!

**Pull request merged**